### PR TITLE
Updating with new config file for zlib 1.2.11

### PR DESCRIPTION
**Pull request by HCLJason**
This is the result of emconfigure.

This needs to get edited and merged after the matching emscripten-ports/zlib PR gets merged.

### CODE REVIEW

**Comment by sbc100** [comment]
Looks good.  Perhaps we could consider moving the config file upstream like I did for sdl2 recently: #14232.   Seems more sensible to keep this file alongside the sources.

Happy to land this change either way.

**Comment by HCLJason** [comment]
kripken mentioned deleting the emscripten-ports version of zlib in the ports PR. It would involve switching to use the official zlib repo of https://github.com/madler/zlib in this file.

**Comment by sbc100** [comment]
> kripken mentioned deleting the emscripten-ports version of zlib in the ports PR. It would involve switching to use the official zlib repo of https://github.com/madler/zlib in this file.

Sure.  I that case I guess we would need to keep the config file here.

In any case, I don't see URL change here is this change.. so either you need to revert the HASH change or update the URL to match the hash.

**Comment by sbc100** [comment]
> > kripken mentioned deleting the emscripten-ports version of zlib in the ports PR. It would involve switching to use the official zlib repo of https://github.com/madler/zlib in this file.
> 
> Sure. I that case I guess we would need to keep the config file here.
> 
> In any case, I don't see URL change here is this change.. so either you need to revert the HASH change or update the URL to match the hash.

As you can see the CI is currently failing because the HASH was changed but the URL wasn't.

File: tools/ports/zlib.py [hunk 1]
```diff
@@ -6,16 +6,16 @@
 import os
 import shutil
 
-TAG = 'version_1'
-HASH = '77f7d8f18fe11bb66a57e358325b7422d721f7b506bd63293cfde74079f958864db66ead5a36c311a76dd8c2b089b7659641a5522de650de0f9e6865782a60dd'
+TAG = 'v1.2.11'
+HASH = '104c62ed1228b5f1199bc037081861576900eb0697a226cafa62a35c4c890b5cb46622e399f9aad82ee5dfb475bae26ae75e2bd6da3d261361b1c8b996970faf'
```
Comment (kripken) [inline tools/ports/zlib.py:10]:
```suggestion
HASH = 'a42b8359e76cf7b3ae70bf31f0f8a8caa407ac80e8fe08b838076cd5e45ac2e685dae45eb59db2d25543fb3b5bd13b843a02bb8373cda704d7238be50d5e9c68'
```

Comment (HCLJason) [reply tools/ports/zlib.py:10]: That's weird. I wonder why it's different.

$ wget https://github.com/madler/zlib/archive/v1.2.11.tar.gz
--2021-05-20 13:24:07--  https://github.com/madler/zlib/archive/v1.2.11.tar.gz
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/madler/zlib/tar.gz/v1.2.11 [following]
--2021-05-20 13:24:08--  https://codeload.github.com/madler/zlib/tar.gz/v1.2.11
Resolving codeload.github.com (codeload.github.com)... 140.82.112.10
Connecting to codeload.github.com (codeload.github.com)|140.82.112.10|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 644492 (629K) [application/x-gzip]
Saving to: ‘v1.2.11.tar.gz’

v1.2.11.tar.gz                100%[=================================================>] 629.39K  --.-KB/s    in 0.1s

2021-05-20 13:24:08 (4.29 MB/s) - ‘v1.2.11.tar.gz’ saved [644492/644492]

$ sha512sum v1.2.11.tar.gz
104c62ed1228b5f1199bc037081861576900eb0697a226cafa62a35c4c890b5cb46622e399f9aad82ee5dfb475bae26ae75e2bd6da3d261361b1c8b996970faf  v1.2.11.tar.gz

**Review by sbc100 — APPROVED** [review]

**Comment by kripken** [comment]
Let me know if you need help debugging the build errors here. (Maybe there is a typo in the tag name, or the folder name needs an update, is my guess.)

**Comment by HCLJason** [comment]
Thanks. I finally got a chance to look at the filename issue, and that's fixed.

It compiles just fine on two different Linux computers for me and the sha512 is consistently a42...

system_libs:INFO: retrieving port: zlib from https://github.com/madler/zlib/archive/v1.2.11.zip
system_libs:INFO: unpacking port: zlib
cache:INFO: generating port: sysroot/lib/wasm32-emscripten/libz.a... (this will be cached in "/local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libz.a" for subsequent builds)
cache:INFO:  - ok

**Review by sbc100 — APPROVED** [review]
Nice!  lgtm.

We are currently experiencing some CI issues.. so if you see failure they could be unrelated to your change.

**Comment by kripken** [comment]
The error here does look like the previous breakage. Merging in latest `main` should fix it.

**Comment by sbc100** [comment]
The diff for this PR seems to includes a lot of changes that already exist in the `main` branch.  I think you may have made some sort of mistake with your rebase or merge operation.

**Pull request closed without merging**